vector/src/main/java/im/vector/app/features/autocomplete/member/AutocompleteMemberController.kt

```diff
@@ -35,12 +37,34 @@ class AutocompleteMemberController @Inject constructor() : TypedEpoxyController<
             return
         }
         val host = this
-        data.forEach { user ->
-            autocompleteMatrixItem {
-                id(user.userId)
-                matrixItem(user.toMatrixItem())
-                avatarRenderer(host.avatarRenderer)
-                clickListener { host.listener?.onItemClick(user) }
+
+        // only separate duplicate displaynames when there are less than 100 on display to save on performance
+        if (data.size > 100) {
+            data.forEach { user ->
+                autocompleteMatrixItem {
+                    id(user.userId)
+                    matrixItem(user.toMatrixItem())
+                    avatarRenderer(host.avatarRenderer)
+                    clickListener { host.listener?.onItemClick(user) }
+                }
+            }
+        } else {
+            val dataSeparated = data.separateDuplicateDisplayNames()
+            dataSeparated.single.forEach { user ->
+                autocompleteMatrixItem {
+                    id(user.userId)
+                    matrixItem(user.toMatrixItem())
+                    avatarRenderer(host.avatarRenderer)
+                    clickListener { host.listener?.onItemClick(user) }
+                }
+            }
+            dataSeparated.duplicate.forEach { user ->
```

Comment:
The order will not be preserved. The users with duplicate display name will always be displayed at the end of the list.